docs/resources/file.md

```diff
@@ -0,0 +1,104 @@
+---
+subcategory: "Storage"
+---
+# databricks_file Resource
+
+This resource allows uploading and downloading files in [databricks_volume](volume.md) up to 2GB in octet-stream.
+
+## Example Usage
+
+In order to manage a file on Unity Catalog Volumes with Terraform, you must specify the `source` attribute containing the full path to the file on the local filesystem.
+
+```hcl
+resource "databricks_catalog" "sandbox" {
+  metastore_id = databricks_metastore.this.id
+  name         = "sandbox"
+  comment      = "this catalog is managed by terraform"
+  properties = {
+    purpose = "testing"
+  }
+}
+
+resource "databricks_schema" "things" {
+  catalog_name = databricks_catalog.sandbox.name
+  name         = "things"
+  comment      = "this schema is managed by terraform"
+  properties = {
+    kind = "various"
+  }
+}
+
+resource "databricks_storage_credential" "external" {
+  name = "creds"
+  aws_iam_role {
+    role_arn = aws_iam_role.external_data_access.arn
+  }
+}
+
+resource "databricks_external_location" "some" {
+  name            = "external-location"
+  url             = "s3://${aws_s3_bucket.external.id}/some"
+  credential_name = databricks_storage_credential.external.name
+}
+
+resource "databricks_volume" "this" {
+  name             = "quickstart_volume"
+  catalog_name     = databricks_catalog.sandbox.name
+  schema_name      = databricks_schema.things.name
+  volume_type      = "EXTERNAL"
+  storage_location = databricks_external_location.some.url
+  comment          = "this volume is managed by terraform"
+}
+
+resource "databricks_file" "this" {
+  source = "/full/path/on/local/system"
+  path   = "/Volumes/${databricks_volume.this.catalog_name}/${databricks_volume.this.schema_name}/{databricks_volume.this.name}/fileName"
+}
+```
+
+You can also inline sources through `content_base64`  attribute.
+
+```hcl
+resource "databricks_file" "init_script" {
+  content_base64 = base64encode(<<-EOT
+    #!/bin/bash
+    echo "Hello World"
+    EOT
+  )
+  path   = "/Volumes/${databricks_volume.this.catalog_name}/${databricks_volume.this.schema_name}/{databricks_volume.this.name}/fileName"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `source` - The full absolute path to the file. Conflicts with `content_base64`.
+* `content_base64` - Contents in base 64 format. Conflicts with `source`.
+* `path` - The path of the file in which you wish to save. Should start with `/Volumes`
```

Comment:
Same, eventually it may not need to start with /Volumes.

@seansmith-db do you know how we should refer to single path namespace in public docs?

Reply:
Right now it's only volumes, right? I would leave it like this until other are supported to avoid customer confusion.

Reply:
Let's say "For example, `/Volumes/main/default/volume1/file.txt`" instead of trying to specify the full set of allowed paths here. We don't have a better reference at this time.

Reply:
Updated